src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -1321,6 +1321,78 @@ def _validate_trusted_launch(namespace):
         namespace.enable_secure_boot = True
 
 
+def _validate_generation_version_and_trusted_launch(cmd, namespace):
+    from azure.cli.core.profiles import ResourceType
+    if not cmd.supported_api_version(resource_type=ResourceType.MGMT_COMPUTE, min_api='2020-12-01'):
+        return
+    from ._vm_utils import trusted_launch_warning_log
+    if namespace.image is not None:
+        from msrestazure.tools import is_valid_resource_id, parse_resource_id
+        from ._vm_utils import is_shared_gallery_image_id, is_community_gallery_image_id,\
+            parse_shared_gallery_image_id, parse_community_gallery_image_id
+
+        if is_shared_gallery_image_id(namespace.image):
+            # create vm with shared gallery
+            from ._client_factory import cf_shared_gallery_image
+            image_info = parse_shared_gallery_image_id(namespace.image)
+            gallery_image_info = cf_shared_gallery_image(cmd.cli_ctx).get(
+                location=namespace.location, gallery_unique_name=image_info[0], gallery_image_name=image_info[1])
+            generation_version = gallery_image_info.hyper_v_generation if hasattr(gallery_image_info,
+                                                                                  'hyper_v_generation') else None
+            features = gallery_image_info.features if hasattr(gallery_image_info, 'features') else None
+            trusted_launch_warning_log(namespace, generation_version, features)
+
+        elif is_community_gallery_image_id(namespace.image):
+            # create vm with community gallery
+            from ._client_factory import cf_community_gallery_image
+            image_info = parse_community_gallery_image_id(namespace.image)
+            gallery_image_info = cf_community_gallery_image(cmd.cli_ctx).get(
+                location=namespace.location, public_gallery_name=image_info[0], gallery_image_name=image_info[1])
+            generation_version = gallery_image_info.hyper_v_generation if hasattr(gallery_image_info,
+                                                                                  'hyper_v_generation') else None
+            features = gallery_image_info.features if hasattr(gallery_image_info, 'features') else None
+            trusted_launch_warning_log(namespace, generation_version, features)
+
+        elif is_valid_resource_id(namespace.image):
+            # create vm with managed image
+            return
+
+        elif urlparse(namespace.image).scheme and "://" in namespace.image:
+            # create vm with vhd
+            return
```

Comment:
Please directly reuse method `_parse_image_argument()` to judge the image type, instead of adding similar logic to increase the maintenance costs